src/test/java/bio/terra/service/common/gcs/GcsUriUtilsTest.java

```diff
@@ -0,0 +1,104 @@
+package bio.terra.service.common.gcs;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.hamcrest.Matchers.equalTo;
+
+import bio.terra.common.category.Unit;
+import bio.terra.service.dataset.exception.InvalidUriException;
+import org.apache.commons.lang3.StringUtils;
+import org.assertj.core.api.Assertions;
+import org.hamcrest.MatcherAssert;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+@Category(Unit.class)
+public class GcsUriUtilsTest {
+
+  @Test
+  public void testGsPathExtraction() {
+    Assertions.assertThat(GcsUriUtils.parseBlobUri("gs://mybucket/my/path.txt"))
+        .usingRecursiveComparison()
```

Comment:
Is this to avoid writing `equals()` for `GsUrlParts`? Using `BlobId` would let us use `equals()` directly.